request.go

```diff
@@ -138,19 +138,49 @@ func (r *Request) close() error {
 			r.cancelCtx()
 		}
 	}()
+
 	r.state.RLock()
 	rd := r.state.readerAt
+	wr := r.state.writerAt
 	r.state.RUnlock()
+
+	var err error
+
 	if c, ok := rd.(io.Closer); ok {
-		return c.Close()
+		if err2 := c.Close(); err == nil {
+			// update error if it is still nil
+			err = err2
+		}
+	}
+
+	if c, ok := wr.(io.Closer); ok {
+		if err2 := c.Close(); err == nil {
+			// update error if it is still nil
+			err = err2
+		}
+	}
```

Comment:
Could you switch those around? Close errors of writers are sometimes data loss, while close errors or readers are mostly not relevant to the user. So I would prefer to get the close error of the writer here.

Reply:
Good point.

Reply:
I have inverted both `Close()` and `transferError()` (to keep them consistent in ordering), and specifically added documentation about why the Writer Close goes first.